### PR TITLE
ruo_full_models

### PR DESCRIPTION
**Pull request by Iva-khar**
Company first level models (full version).
Migrate after merging!
Added constraint to company table

### CODE REVIEW

File: data_ocean/models/company_models.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+class CompanyDetail(DataOceanModel):
+    company = models.ForeignKey(Company, on_delete=models.CASCADE)
+    founding_document_num = models.CharField(max_length=100, null=True)
+    executive_power = models.CharField(max_length=100, null=True)
+    superior_management = models.CharField(max_length=100, null=True)
+    authorized_capital = models.CharField(max_length=100, null=True)
+    managing_paper = models.CharField(max_length=100, null=True)
+    terminated_info = models.CharField(max_length=100, null=True)
+    termination_cancel_info = models.CharField(max_length=100, null=True)
+    vp_dates = models.CharField(max_length=100, null=True)
```
Comment (Baranovsky1986) [inline data_ocean/models/company_models.py:58]: as i know we don`t need use def__str__, cose it comes from abstract model DataOceanModel

Comment (Iva-khar) [reply data_ocean/models/company_models.py:58]: yes )

File: data_ocean/models/company_models.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        return f"{self.name}, код: {self.code}"
+
+class CompanyType(DataOceanModel):
+    name = models.CharField(max_length=100, unique=True, null=True)
+    
+    def __str__(self):
+        return self.name
+
+class Bylaw(DataOceanModel):
+    name = models.CharField(max_length=100, unique=True, null=True)
```
Comment (criticus) [inline data_ocean/models/company_models.py:20]: could it be null?

File: data_ocean/models/company_models.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+class Authority(DataOceanModel):
+    name = models.CharField(max_length=200, unique=True, null=True)
+    code = models.CharField(max_length=10, unique=True, null=True)
+    
+    def __str__(self):
+        return f"{self.name}, код: {self.code}"
+
+class CompanyType(DataOceanModel):
+    name = models.CharField(max_length=100, unique=True, null=True)
```
Comment (criticus) [inline data_ocean/models/company_models.py:14]: could it be null?

File: data_ocean/models/company_models.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
+from data_ocean.models.main import DataOceanModel
+from data_ocean.models.kved_models import Kved
+from data_ocean.models.ruo_models import State
+from django.db import models
+
+class Authority(DataOceanModel):
+    name = models.CharField(max_length=200, unique=True, null=True)
+    code = models.CharField(max_length=10, unique=True, null=True)
```
Comment (criticus) [inline data_ocean/models/company_models.py:8]: null?

File: data_ocean/models/company_models.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    name = models.CharField(max_length=100, unique=True, null=True)
+    
+    def __str__(self):
+        return self.name
+
+class Company(DataOceanModel):
+    INVALID = 'invalid'
+    name = models.CharField(max_length=500, null=True)
+    short_name = models.CharField(max_length=500, null=True)
+    company_type = models.ForeignKey(CompanyType, on_delete=models.CASCADE)
```
Comment (criticus) [inline data_ocean/models/company_models.py:29]: `ALTER TABLE my_table ADD CONSTRAINT my_constraint CHECK (
  (column_1 IS NULL) != (column_2 IS NULL));`

File: data_ocean/models/company_models.py (hunk 1 above)
Comment (Iva-khar) [reply data_ocean/models/company_models.py:46]: fixed

File: data_ocean/models/company_models.py (hunk 2 above)
Comment (Iva-khar) [reply data_ocean/models/company_models.py:14]: We don't know

**Comment by Iva-khar** [comment]
```
IegorSoboliev left a comment
Maybe we can have clearer naming for 'vp_dates'?
```
We don't know what it is

File: data_ocean/models/company_models.py (hunk 5 above)
Comment (Iva-khar) [reply data_ocean/models/company_models.py:20]: is it Django?

File: data_ocean/models/company_models.py (hunk 3 above)
Comment (Iva-khar) [reply data_ocean/models/company_models.py:11]: Well, what if it does?

File: data_ocean/models/company_models.py (hunk 4 above)
Comment (Iva-khar) [reply data_ocean/models/company_models.py:8]: fixed with 'name' field. 'code' field may be null when it is current_authority

File: data_ocean/models/company_models.py (hunk 3 above)
Comment (Iva-khar) [reply data_ocean/models/company_models.py:11]: fixed

File: data_ocean/models/company_models.py (hunk 5 above)
Comment (Iva-khar) [reply data_ocean/models/company_models.py:20]: done in migration#44

Comment (Vital-jan) [reply data_ocean/models/company_models.py:20]: Could it need "cascade"? on_delete=models.CASCADE

Comment (Iva-khar) [reply data_ocean/models/company_models.py]: it makes sense... done.

**Comment by criticus** [comment]
@Iva-khar do you still want to merge this?

**Comment by Iva-khar** [comment]
criticus commented 5 days ago
@Iva-khar do you still want to merge this?

Yes

**Pull request merged**